source/com.microsoft.tfs.client.common.ui/src/com/microsoft/tfs/client/common/ui/controls/connect/ServerTypeSelectControl.java

```diff
@@ -25,25 +35,32 @@
 import com.microsoft.tfs.client.common.ui.controls.generic.BaseControl;
 import com.microsoft.tfs.client.common.ui.controls.generic.compatibility.link.CompatibilityLinkControl;
 import com.microsoft.tfs.client.common.ui.controls.generic.compatibility.link.CompatibilityLinkFactory;
+import com.microsoft.tfs.client.common.ui.dialogs.connect.ServerListDialog;
 import com.microsoft.tfs.client.common.ui.framework.command.UICommandExecutorFactory;
 import com.microsoft.tfs.client.common.ui.framework.command.UICommandFinishedCallbackFactory;
 import com.microsoft.tfs.client.common.ui.framework.helper.SWTUtil;
 import com.microsoft.tfs.client.common.ui.framework.layout.GridDataBuilder;
+import com.microsoft.tfs.core.config.persistence.DefaultPersistenceStoreProvider;
 import com.microsoft.tfs.core.util.ServerURIUtils;
 import com.microsoft.tfs.core.util.URIUtils;
+import com.microsoft.tfs.core.util.serverlist.ServerList;
+import com.microsoft.tfs.core.util.serverlist.ServerListConfigurationEntry;
+import com.microsoft.tfs.core.util.serverlist.ServerListManagerFactory;
 import com.microsoft.tfs.util.listeners.SingleListenerFacade;
 
 public class ServerTypeSelectControl extends BaseControl {
     private static final Log log = LogFactory.getLog(ServerTypeSelectControl.class);
 
     private URI serverURI = null;
-    private boolean serverReadonly = false;
     private boolean ignoreServerChangeEvents = false;
     private ICommandExecutor commandExecutor;
     private ICommandExecutor noErrorDialogCommandExecutor;

```

Comment:
We do not use these executors. I remember, we already discussed some similar case, but do not remember the decision, I'm sorry to say. :-)